src/components/FileInputField/FileInputField.jsx

```diff
@@ -17,49 +25,142 @@ export const FileInputField = React.forwardRef((props, ref) => {
     isLabelVisible,
     label,
     layout,
+    onFilesChanged,
     required,
+    size,
     validationState,
     validationText,
     ...restProps
   } = props;
 
-  const context = useContext(FormLayoutContext);
+  const formLayoutContext = useContext(FormLayoutContext);
+  const inputGroupContext = useContext(InputGroupContext);
+  const translations = useContext(TranslationsContext);
+
+  const [selectedFileNames, setSelectedFileNames] = useState([]);
+  const [isDragAndDropSupported, setIsDragAndDropSupported] = useState(false);
+  const [isDragging, setIsDragging] = useState(false);
+
+  const handleFileChange = (files) => {
+    if (files.length === 0) {
+      setSelectedFileNames([]);
+      return;
+    }
+
+    const fileNames = [];
+
+    [...files].forEach((file) => {
+      fileNames.push(file.name);
+    });
+
+    setSelectedFileNames(fileNames);
+  };
+
+  const handleInputChange = (event) => {
+    handleFileChange(event.target.files);
+
+    if (props?.onFilesChanged) {
+      props.onFilesChanged(event);
+    }
+  };
+
+  const handleClick = (event) => {
+    event.currentTarget.previousElementSibling.click();
+  };
+
+  const handleDrop = (event) => {
+    event.preventDefault();
+    handleFileChange(event.dataTransfer.files);
```

Comment:
`handleDrop` does not call `props.onFilesChanged`, so this is reason why it does not work on drop.
Please, handle both events as I requested earlier this day and pass event itself and files, two parameters.

props.onFilesChanged(files, e) or props.onFilesChanged(e, files), i do not care which way. Also `onFilesChanged` should be required in my opinion, there is no way to work with files uncontrolled way.